### PR TITLE
Add type hints to most files, and add to CI

### PR DESCRIPTION
**Pull request by berland**

### CODE REVIEW

File: .github/workflows/subscript.yml [hunk 1]
```diff
@@ -46,6 +46,11 @@ jobs:
           pip install flake8
           flake8 src tests
 
+      - name: Enforce static typing
+        run: |
+          pip install mypy
+          mypy src/subscript
```
Comment (jondequinor) [inline .github/workflows/subscript.yml:52]: Why not add this to test_requirements.txt or similar?

File: src/subscript/pack_sim/pack_sim.py [hunk 2]
```diff
@@ -420,50 +434,72 @@ def inspect_file(
...
-        elif line_strip == "OPTIMIZE" and fmu_include:
-            section = "optimize/"
-            if not os.path.exists("%s/include/%s" % (packing_path, section)):
-                os.makedirs("%s/include/%s" % (packing_path, section))
+        elif line_strip == "OPTIMIZE" and fmu_include:  # type: ignore
+            section = "optimize/"  # type: ignore
+            if not os.path.exists(
+                "%s/include/%s" % (packing_path, section)  # type: ignore
+            ):
+                os.makedirs("%s/include/%s" % (packing_path, section))  # type: ignore
```
Comment (jondequinor) [inline src/subscript/pack_sim/pack_sim.py:502]: What's going on here? And could this maybe be better handled by ignoring the file in a mypy config?

File: .github/workflows/subscript.yml (hunk 1 above)
Comment (jondequinor) [reply .github/workflows/subscript.yml:52]: OKay, I see it was added (at the bottom).

File: src/subscript/pack_sim/pack_sim.py (hunk 2 above)
Comment (berland) [reply src/subscript/pack_sim/pack_sim.py:502]: You've found the black sheep in subscript.. `pack_sim` is in need for a general rewrite yes, (see #363). I struggled a bit with this part but found no better way than ignoring these lines. Ignoring the whole file is perhaps a little bit too permissive?

Comment (jondequinor) [reply src/subscript/pack_sim/pack_sim.py:502]: This is cluttering the history, so I lean towards ignoring the whole file.

Comment (berland) [reply src/subscript/pack_sim/pack_sim.py:502]: Will fix 👍🏻

**Review by jondequinor — APPROVED** [review]
Skimmed most parts, looked closely on some. Overall I think this is good after the excessive `# type: ignore` fix.

**Pull request merged**